libraries/lib-exceptions/AudacityException.h

```diff
@@ -226,7 +228,7 @@ noexcept(
          // At this point, e is the "current" exception, but not "uncaught"
          // unless it was rethrown by handler.  handler might also throw some
          // other exception object.
-         if (!std::uncaught_exception()) {
+         if (!HasUncaughtExceptions()) {
```

Comment:
If `uncaught_exceptions` is available, then a more correct test would be this:

- save its value before the `finally`
- test here that its current value is not more than the saved

Reply:
Fixed and force-pushed